public/app/plugins/datasource/cloudwatch/ConfigEditor.tsx

```diff
@@ -0,0 +1,389 @@
+        if (config.hasOwnProperty('secureJsonData') && config.secureJsonData.hasOwnProperty(l)) {
+          delete config.secureJsonData[l];
+        }
+      } else {
+        config.secureJsonData[l] = config.editorSecureJsonData[l];
+      }
+    }
+
+    this.props.onOptionsChange({
+      ...config,
+    });
+  };
+
+  onAuthProviderChange = (authType: SelectableValue<string>) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorJsonData: {
+        ...this.state.config.editorJsonData,
+        authType: authType.value,
+      },
+    });
+  };
+
+  onRegionChange = (defaultRegion: SelectableValue<string>) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorJsonData: {
+        ...this.state.config.editorJsonData,
+        defaultRegion: defaultRegion.value,
+      },
+    });
+  };
+
+  onResetAccessKey = () => {
+    this.updateDatasource({
+      ...this.state.config,
+      version: this.state.config.version + 1,
+      secureJsonFields: {
+        ...this.state.config.secureJsonFields,
+        accessKey: false,
+      },
+    });
+  };
+
+  onAccessKeyChange = (accessKey: string) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorSecureJsonData: {
+        ...this.state.config.editorSecureJsonData,
+        accessKey,
+      },
+    });
+  };
+
+  onResetSecretKey = () => {
+    this.updateDatasource({
+      ...this.state.config,
+      version: this.state.config.version + 1,
+      secureJsonFields: {
+        ...this.state.config.secureJsonFields,
+        secretKey: false,
+      },
+    });
+  };
+
+  onSecretKeyChange = (secretKey: string) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorSecureJsonData: {
+        ...this.state.config.editorSecureJsonData,
+        secretKey,
+      },
+    });
+  };
+
+  onCredentialProfileNameChange = (database: string) => {
+    this.updateDatasource({
+      ...this.state.config,
+      database,
+    });
+  };
+
+  onArnAssumeRoleChange = (assumeRoleArn: string) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorJsonData: {
+        ...this.state.config.editorJsonData,
+        assumeRoleArn,
+      },
+    });
+  };
+
+  onCustomMetricsNamespacesChange = (customMetricsNamespaces: string) => {
+    this.updateDatasource({
+      ...this.state.config,
+      editorJsonData: {
+        ...this.state.config.editorJsonData,
+        customMetricsNamespaces,
+      },
+    });
+  };
+
+  render() {
+    const { config, authProviderOptions, regions } = this.state;
+
+    return (
+      <>
+        <h3 className="page-heading">CloudWatch Details</h3>
+        <div className="gf-form-group">
+          <div className="gf-form-inline">
+            <div className="gf-form">
+              <FormLabel className="width-14" tooltip="Choose an Azure Cloud.">
+                Auth Provider
+              </FormLabel>
+              <Select
+                className="width-15"
```

Comment:
Doesn't it look better if all fields are the same width? So the inputs also have a width of 30, like the custom metric, access key etc?

Reply:
I guess so? 🤷🏽‍♀️